lib/users/lms/get-lms-link.js

```diff
@@ -0,0 +1,182 @@
+ * with the courses corresponding to its userTypes
+ * @param  {Object}   user
+ * @return {Object}   url  contains the link to redirect to consume the SQSSO
+ */
+function getLMSLink (args, cb) {
+  var seneca = this;
+  var plugin = args.role;
+  var APIKey = process.env.LMS_KEY;
+  var user = args.user;
+  var APIUrl = 'https://coderdojo.learnupon.com/api/v1/';
+  var hash = crypto.createHash('md5');
+  var response = {
+    'url': ''
+  };
+  var LMSUsername = process.env.LMSUsername;
+  var LMSPassword = process.env.LMSPassword;
+
+  /**
+   * GetLMSUser or Create it if it doesn't exists
+   * @return {Number} lmsUserId
+   * @return {Array} UserTypes as GroupMembership
+   */
+  function getUser (waterfallCb) {
+    var userTypes = [];
+    var allowedUserTypes = ['mentor', 'champion'];
+    seneca.act({role: 'cd-profiles', cmd: 'load_user_profile', userId: user.id},
+      function (err, profile) {
+        userTypes = [profile.userType];
+        //  Get extended userTypes
+        seneca.act({role: 'cd-dojos', cmd: 'load_usersdojos', query:{ userId: user.id}},
+          function (err, userDojos) {
+            userTypes = _.flatten(userTypes.concat(_.map(userDojos, 'userTypes')));
+            userTypes = _.intersection(userTypes, allowedUserTypes);
+            console.log('inters', userTypes, allowedUserTypes);
+            if (!_.isEmpty(userTypes)) {
+              if (_.isEmpty(user.lmsId)) {
+                //  The user doesn't exists yet on the LMS, we create it to save the corresponding Id
+                 request.post(APIUrl + 'users', {
+                  auth: {
+                    user: LMSUsername,
+                    pass: LMSPassword
+                  },
+                  json: {
+                    User: {
+                      email: user.email,
+                      password: Uuid() }
+                    }
+                  }, function(err, res, lmsUser) {
+                    seneca.act({role: 'cd-users', cmd: 'update', user: {
+                      id: user.id,
+                      lmsId: lmsUser.id
+                    }}, function (err, profile) {
+                      waterfallCb(null, lmsUser.id, userTypes);
+                    });
+                  });
+              } else {
+                //  The user already exists in the LMS, no need to recreate it
+                waterfallCb(null, user.lmsId, userTypes);
+              }
+            } else {
+              cb(null, {ok: false, why: 'UserType not allowed', http$: { status: 403}});
+            }
+        });
+
+    });
+  }
+
+  /**
+   * Recover actual membership on LMS
+   * @param  {Number} lmsUserId
+   * @param  {Array} userTypes
+   * @param  {Fn} waterfallCb
+   */
+  function getUserMemberships (lmsUserId, userTypes, waterfallCb) {
+    // Get the user in LearnUpon
+    request.get(APIUrl + 'group_memberships?user_id=' + lmsUserId, {
+      auth: {
+        user: LMSUsername,
+        pass: LMSPassword
+      },
+      json: {
```

Comment:
I don't think this is needed? Same info is in the URL, which is what the docs say to do.